cmd/cmdutil/version.go

```diff
@@ -0,0 +1,52 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package cmdutil
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+)
+
+// SnapdVersionFromInfoFile returns snapd version read for the
+// given info" file, pointed by infoPath.
+// The format of the "info" file is a single line with "VERSION=..."
+// in it. The file is produced by mkversion.sh and normally installed
+// along snapd binary in /usr/lib/snapd.
+func SnapdVersionFromInfoFile(infoPath string) (string, error) {
+	content, err := ioutil.ReadFile(infoPath)
+	if err != nil {
+		return "", fmt.Errorf("cannot open snapd info file %q: %s", infoPath, err)
+	}
+
+	if !bytes.HasPrefix(content, []byte("VERSION=")) {
```

Comment:
This is a little bit complex for what it needs to be. Can we just use a key=value parser we probably have implemented a few times to find VERSION and just return that?

Reply:
Agree, but this is just moving the old code around, I deliberately kept it as is. I'd leave any refactoring for a followup.